docs/ADR-Resize-Channel-Security.md

```diff
@@ -0,0 +1,130 @@
+
+1. User calls the Clearnode API with a resize request
+2. Clearnode validates the request and user balance
+3. Clearnode generates and signs a new state with RESIZE intent
+4. Clearnode returns the signed state to the user
+5. User signs the state and submits it to the smart contract
+
+This process has revealed a vulnerability: users can double-spend their funds by requesting resizes across multiple chains that collectively exceed their total balance. Since the server signs states that remain valid indefinitely and doesn't track pending state commitments across chains, a user could:
+
+1. Deposit 5 USDC on Chain 1 and 5 USDC on Chain 2 (total balance of 10 USDC)
+2. Request a resize of Channel 1 from 5 to 10 USDC (server signs this state)
+3. Request a resize of Channel 2 from 5 to 10 USDC (server signs this state too)
+4. Submit both states to their respective blockchains
+5. Result: User has committed 20 USDC despite only having 10 USDC in their balance
+
+## Decision Drivers
+* Security: Prevent double-spending across multiple chains
+* User Experience: Maintain a seamless resize process
+* System Integrity: Ensure balance accounting remains accurate
+* Technical Feasibility: Implement with minimal changes to existing contracts
+* Operational Overhead: Consider the impact on server operations
+
+## Considered Options
+
+### Option 1: Timeout-Based Fund Locking
+* Lock funds in the database when a resize is requested
+* Set a timeout to release locks if not confirmed on-chain
+* Challenge: The smart contract has no notion of state expiration, so users could still submit expired states
+
+### Option 2: State Versioning with Registry
+* Maintain a server-side registry of valid state versions
+* When signing a new state, include a global version number in state data
+* When processing on-chain events, invalidate all previous versions
+* Challenge: **Critical flaw** - Once a state is signed by the server, it can be submitted to the blockchain regardless of database state. Without smart contract modifications to validate against the external registry, this approach only detects double-spending after it occurs but doesn't prevent it.
+
+### Option 3: Cross-Chain Lock Commitments
+* Modify smart contracts to require a "lock" transaction on a primary chain
+* This lock creates a verifiable on-chain commitment of total funds allocation
+* Other chains verify against this commitment before processing resizes
+* Each resize reduces the available commitment amount
+* Challenge: Requires smart contract modifications, adds cross-chain dependencies, and significantly complicates the user experience with additional transactions
+
+### Option 4: Server-Side Blockchain Submission
+* Instead of returning signed states to users, server directly submits resize transactions
```

Comment:
`resize` state still needs to be unanimously signed, therefore the flow is the following:
- server composes the state
- server sends it to the user to be signed
- user sends the signature back 
- server countersigns and sends tx to chain

Reply:
I also think that this approach is worse then (1), because we hide valid states from users (no data availability). If server goes down, user will not be able to withdraw funds (because they do not possess the `resize` state). Moreover, if we commit to data availability now or in the future, we will still need to implement the `getLatestSupportedState` on the broker side. If we go with (4), this function would need to have an exception and do NOT send `resize` state if it is the latest, which introduces inconsistency.